libs/gitignore/Gitignore_filter.ml

```diff
@@ -99,8 +106,13 @@ let select_path opt_gitignore_file_cache sel_events levels relative_segments =
 let select t sel_events (full_git_path : Ppath.t) =
   let rel_segments =
     match Ppath.segments full_git_path with
+    (* Remove empty line segment *)
     | "" :: xs -> xs
-    | __else__ -> assert false
+    | "/" :: _ ->
+        failwith
+          (Printf.sprintf "Gitignore: full_git_path %s not relative"
+             Ppath.(to_string full_git_path))
```

Comment:
Does this case indicate a bug in the program? If so, we should use the `Invalid_argument` exception which is raised with `invalid_arg` instead of `failwith`. It's like `assert false` but shifts the blame to the caller of the function. `failwith` (the `Failure` exception) is more when you're lazy and didn't bother to classify the error.

Reply:
This was me trying to get a better error message instead of `assert false`. This was my interpretation of why this assert false existed before, but I wasn't sure. I can switch to Invalid_argument if you think that makes more sense, but I wasn't sure if that was the intention of the original assert false.